packages/adapters-library/src/core/apy-calculators/ConstantProductAmmApyCalculator.ts

```diff
@@ -0,0 +1,136 @@
+import { Underlying } from '../../types/adapter'
+import { NotSupportedError } from '../errors/errors'
+import { BigDecimal } from '../utils/BigDecimal'
+import { collectLeafTokens } from '../utils/collectLeafTokens'
+import { GetApyArgs } from './ApyCalculator'
+import { BaseApyCalculator } from './BaseApyCalculator'
+
+/**
+ * Calculates the APY of a position in a constant product AMM like Uniswap V2.
+ * It isolates the fees earned from impermanent loss by calculating adjusted balances:
+ *
+ * balance(t') = balance(t) + fees(t) + impermanentLoss(t)
+ *
+ * where:
+ *
+ * balance(t) + impermanentLoss(t) = adjustedBalance(t')
+ *
+ * Compatible protocols / products are:
+ * - Constant Product AMMs (Uniswap V2, SushiSwap, ...)
+ */
+export class ConstantProductAmmApyCalculator extends BaseApyCalculator {
```

Comment:
Formula is detailed in [this Google Docs](https://docs.google.com/document/d/1n1kCpdNf81UiYBLBwgd98ESNIf-Yx86unnsHOFIFNtI/edit?tab=t.0)